clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/CloudMetricStatistics.groovy

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import groovy.transform.Immutable
+
+interface CloudMetricStatistics {
+  /**
+   * Unit of measurement for all datapoints; should be the plural form of the unit if applicable,
+   * e.g. "Bytes", "Percent", "Kilobytes/Second"
+   */
+  String unit
+
+  /**
+   * List of statistical datapoints; at least one statistic (average, sum, sampleCount, minimum, maximum) should be
+   * populated, as well as the timestamp
+   */
+  List<CloudMetricDatapoint> datapoints
+
+  @Immutable
```

Comment:
I wonder if this should be an interface rather than a data object. I am assuming performance doesnt matter. An interface could just wrap your AWS data so these methods are straight through. When you construct the object, you just pass the data instance through and bind it to your class that implements the interface and returns the appropriate fields instead of breaking out the fields.

The benefit is now other platforms can come through and possibly add their additional data and have the option to compute some of it on demand rather than eagerly transform it.

Just a thought. We can refactor it later after adding another provider.